node/tests/ServerModules.test.ts

```diff
@@ -57,4 +62,166 @@ describe("GlideJson", () => {
             expect(info).toContain("# search_index_stats");
         },
     );
+
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        "ServerModules json.set and json.get tests",
+        async (protocol) => {
+            client = await GlideClusterClient.createClient(
+                getClientConfigurationOption(cluster.getAddresses(), protocol),
+            );
+            const key = uuidv4();
+            const jsonValue = { a: 1.0, b: 2 };
+
+            // JSON.set
+            expect(
+                await GlideJson.set(
+                    client,
+                    key,
+                    "$",
+                    JSON.stringify(jsonValue),
+                ),
+            ).toBe("OK");
+
+            // JSON.get
+            let result = await GlideJson.get(client, key, { paths: ["."] });
+            expect(JSON.parse(result.toString())).toEqual(jsonValue);
+
+            // JSON.get with array of paths
+            result = await GlideJson.get(client, key, {
+                paths: ["$.a", "$.b"],
+            });
+            expect(JSON.parse(result.toString())).toEqual({
+                "$.a": [1.0],
+                "$.b": [2],
+            });
+
+            // JSON.get with non-existing key
+            expect(
+                await GlideJson.get(client, "non_existing_key", {
+                    paths: ["$"],
+                }),
+            );
+
+            // JSON.get with non-existing path
+            result = await GlideJson.get(client, key, { paths: ["$.d"] });
+            expect(result).toEqual("[]");
+        },
+    );
+
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
+        "ServerModules json.set and json.get tests with multiple value",
+        async (protocol) => {
+            client = await GlideClusterClient.createClient(
+                getClientConfigurationOption(cluster.getAddresses(), protocol),
+            );
+            const key = uuidv4();
+
+            // JSON.set with complex object
+            expect(
+                await GlideJson.set(
+                    client,
+                    key,
+                    "$",
+                    JSON.stringify({ a: { c: 1, d: 4 }, b: { c: 2 }, c: true }),
+                ),
+            ).toBe("OK");
+
+            // JSON.get with deep path
+            let result = await GlideJson.get(client, key, { paths: ["$..c"] });
+            expect(JSON.parse(result.toString())).toEqual([true, 1, 2]);
```

Comment:
are we worried about the order of this result?

Reply:
JSON.GET documentation does not mention the order of results when a single path is passed, I assume the order should be consistent since other clients have equivalent test methods.

Reply:
expect.toEqual() peforms a deep equality check actually.